### PR TITLE
ref(global-project-configs): Add generic filters placeholders

### PR DESCRIPTION
**Pull request by iker-barriocanal**
This PR adds placeholders to add generic filters to global and project configs.

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -170,9 +171,26 @@ def get_filter_settings(project: Project) -> Mapping[str, Any]:
     if csp_disallowed_sources:
         filter_settings["csp"] = {"disallowedSources": csp_disallowed_sources}
 
+    try:
+        generic_filters = _get_generic_project_filters()
+    except Exception:
+        logger.exception(
+            "Exception while building Relay project config: error building generic filters"
+        )
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:179]: Is this better than `sentry_sdk.capture_exception()`?

File: src/sentry/relay/globalconfig.py [hunk 2]
```diff
@@ -16,11 +17,34 @@
...
+    condition: Mapping[str, str] | None
+    """A rule condition in the DSL compatible with Relay.
+
+    See https://github.com/getsentry/relay/blob/d4b8402e6853eb62b2402f8f8c8482adae518474/relay-protocol/src/condition.rs#L341.
+    """
+
+
+class GenericFiltersConfig(TypedDict):
+    version: int
+    filters: Sequence[GenericFilter]
```
Comment (jjbayer) [inline src/sentry/relay/globalconfig.py:32]: Should this be defined in `relay.config` instead of `relay.globalconfig`?

**Review by jjbayer — COMMENTED** [review]
Any way we can test this?

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/relay/config/__init__.py:178]: Not sure if it's better, but these exceptions should also reach S4S.

File: tests/sentry/api/endpoints/test_relay_globalconfig_v3.py [hunk 3]
```diff
@@ -78,3 +78,26 @@ def test_return_global_config_on_right_version(
...
+                    "value": "Firefox",
+                },
+            }
+        ],
+    },
+)
+@patch("sentry.relay.globalconfig.RELAY_OPTIONS", [])
+def test_global_config_valid_with_generic_filters():
+    config = get_global_config()
+    assert config == normalize_global_config(config)
```
Comment (jjbayer) [inline tests/sentry/api/endpoints/test_relay_globalconfig_v3.py:103]: Nice! I would add a test like this for project configs as well, to be safe.

File: src/sentry/relay/config/__init__.py [hunk 4]
```diff
@@ -170,9 +170,41 @@ def get_filter_settings(project: Project) -> Mapping[str, Any]:
...
+        if generic_filters and len(generic_filters["filters"]) > 0:
+            filter_settings["generic"] = generic_filters
+
     return filter_settings
 
 
+class GenericFilter(TypedDict):
+    id: str
+    isEnabled: bool
+    condition: Mapping[str, str] | None
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:189]: Can we type this out? There is already types for this in https://github.com/getsentry/sentry/blob/38a91b6dc20655fa299a6b2a13b10231d9480c39/src/sentry/dynamic_sampling/rules/utils.py#L95-L110 (might need to be extended).

**Review by jjbayer — APPROVED** [review]

File: tests/sentry/api/endpoints/test_relay_globalconfig_v3.py (hunk 3 above)
Comment (iker-barriocanal) [reply tests/sentry/api/endpoints/test_relay_globalconfig_v3.py:106]: lol, I must have discarded it dealing with git. Should be back in [5031c20](https://github.com/getsentry/sentry/pull/66410/commits/5031c2003dd80040e3cc4c9dda440b2ff6471ee7).

**Pull request merged**